src/Builder/Encoder/OperatorEncoder.php

```diff
@@ -129,6 +132,34 @@ private function encodeAsObject(OperatorInterface $value): stdClass
             : $this->wrap($value, $result);
     }
 
+    private function encodeAsSearch(OperatorInterface $value): stdClass
+    {
+        assert($value instanceof SearchStage || $value instanceof SearchMetaStage);
+
+        $result = new stdClass();
+        foreach (get_object_vars($value) as $key => $val) {
+            // Skip optional arguments. If they have a default value, it is resolved by the server.
+            if ($val === Optional::Undefined) {
+                continue;
+            }
+
+            // Merge the operator properties into the result object
+            if ($key === 'operator') {
+                $val = $this->recursiveEncode($val);
+                assert($val instanceof stdClass);
+                foreach (get_object_vars($val) as $subkey => $subval) {
```

Comment:
If `$val` is a stdClass, do you even need `get_object_vars()`? [Basic object iteration](https://www.php.net/manual/en/language.oop5.iterations.php) rules should apply. No objections if you feel this is more explicit/consistent.

Separate question: is the logic here also frequently applied in other operators? Or is it unique in its unwrapping and merging of the `operator` object into the parent context?

Is there any concern that operator keys (i.e. `$subkey` values) might conflict with a top-level `$key`?

Reply:
> If `$val` is a stdClass, do you even need `get_object_vars()`? [Basic object iteration](https://www.php.net/manual/en/language.oop5.iterations.php) rules should apply. No objections if you feel this is more explicit/consistent.

For the time being, in removed the `get_object_vars`.

> Separate question: is the logic here also frequently applied in other operators? Or is it unique in its unwrapping and merging of the `operator` object into the parent context?
> Is there any concern that operator keys (i.e. `$subkey` values) might conflict with a top-level `$key`?

I opened a PR to solve the need of merging the subproperties into the parent object: https://github.com/mongodb/mongo-php-library/pull/1540